src/Rings/MPolyQuo.jl

```diff
@@ -1,6 +1,7 @@
 export singular_poly_ring, singular_coeff_ring, MPolyQuo, MPolyQuoElem, MPolyQuoIdeal
 export quo, base_ring, modulus, gens, ngens, dim, simplify!, default_ordering
 export issubset
+export is_prime, pre_image_ideal
```

Comment:
`is_prime` is already exported so why export again?

Reply:
Further I do not think that `pre_image_ideal` should be exported. 
The terminology is general and ambiguous. 
For integers in $\ZZ/n \ZZ$ we have lift(x) to return a representative. 

But I think a better option is to consequently work with morphisms as we do in mathematics. 
I suggest the following
```
julia> P,(x,y) = PolynomialRing(QQ,[:x,:y])
(Multivariate Polynomial Ring in x, y over Rational Field, fmpq_mpoly[x, y])

julia> R, phi = quo(P,ideal(P,x))
(Quotient of Multivariate Polynomial Ring in x, y over Rational Field by ideal(x), Map from
Multivariate Polynomial Ring in x, y over Rational Field to R defined by a julia-function with inverse)

julia> kernel(phi)
ERROR: MethodError: no method matching kernel(::MapFromFunc{FmpqMPolyRing, MPolyQuo{fmpq_mpoly}})

julia> preimage(phi, J)
ERROR: MethodError: no method matching parent(::MPolyQuoIdeal{fmpq_mpoly})
```
This is the mathematical way to do it. Hence it should be the Oscar way. 

As the example shows, unfortunately, phi does not know that it is a homomorphism.
It should know that! One might say that an actual kernel computation is too expensive here.
But that is just a matter of implementation and this is a matter of the interface. 
For instance one could give `phi` a type `ResidueRingHomomorphism`  and create a method
`preimage(phi::ResiedueRingHomomorphism, I::MPolyQuoIdeal) = pre_image_ideal(I)`

Or let me say that I can live with `pre_image_ideal` being exported 
but only after the example above works. 
... but perhaps this rather belongs in an issue?

Reply:
In general, we can expect maps like `phi` above to only be `Hecke.Map`s without any further information or type. In many cases, there are no special types for whatever they are supposed to do and they are created via `MapFromFunc`. So relying on implementations of a `kernel` routine seems to be a rather long shot here and should be discussed elsewhere. 

The `pre_image_ideal` was never really thought to be exported. It was an internal of the localization stuff and just popped up again in the line added by @afkafkafk13 for `MPolyQuoIdeal`s. 

I suggest to just take the direct way for the time being and discuss any further ensuing questions more broadly.

Reply:
In my opinion Simon's point of view makes more sense and should be the way to go. 

I only used pre_image_ideal for two reasons at the other place:
1. the structurally correct approach through the map did not work 
    (whence I assumed it was not meant to be used here)
3. the pre_image_ideal method was already used in the seemingly analogous localized case
    (whence I assumed it to be the preferred method)

Reply:
> `is_prime` is already exported so why export again?

It is an indicator in the header of the file what is implemented below. It doesn't hurt to have this, plus: I repeatedly encountered problems because export statements occured only once and were removed by someone doing completely unrelated stuff (e.g. just moving files or something), probably thinking the same: "Ha! This is rather general, so it will probably not hurt to remove this and this shouldn't be here in the first place." and then other things suddenly crashed. 

My personal bottom line: Better have more exports. 

Same applies to the new changes.

Reply:
> In my opinion Simon's point of view makes more sense and should be the way to go.

But can you live with the current solution now? 

I mean: This is just a bug-fix. We should not make this particular PR the place to discuss fundamental design decisions in general. We can do that, but please, let's just get this stuff here working again, first.

Reply:
The problem is that this makes it rather hard to rename or remove exported symbols. Now they are scattered everywhere and there is no systematic way to find them.

Reply:
I already approved this in the spirit of a bug fix and I still keep my approval.

But I wanted to answer Simon pointing out that we need to discuss this in general (and in person!) and then implement the chosen structural solution afterwards

Reply:
If this is just a bug-fix, then the export should not have been added.

Reply:
> The problem is that this makes it rather hard to rename or remove exported symbols. Now they are scattered everywhere and there is no systematic way to find them.

I see. Alright, then I can step back from this philosophy.